gamemodes/terrortown/gamemode/server/sv_corpse.lua

```diff
@@ -732,3 +732,12 @@ function GM:TTT2ModifyRagdollVelocity(deadply, rag, velocity) end
 -- @hook
 -- @realm server
 function GM:TTTOnCorpseCreated(rag, deadply) end
+
+---
+-- Called after a player has been given credits for searching a corpse.
+-- @param Player ply The player that searched the corpse
+-- @param Entity rag The ragdoll that was searched
+-- @param number credits The amount of credits that were given
+-- @hook
+-- @realm server
+function GM:TTT2OnGiveFoundCredits(ply, rag, credits) end
```

Comment:
This hook seems to be called no where?

Reply:
I missed in merging from upstream that the place where this was called moved from sv_corpse to bodysearch. I've added the call back and will test sometime in the next few days.

Reply:
I'm not sure that adding the additional functionality you've suggested would work for my stats addon, like you said.

The event system currently doesn't have events for the `GM:TTT2OnReceiveKillCredits()`, `GM:TTT2OnReceiveTeamAwardCredits()` or `GM:TTT2OnTransferCredits()` cases. If you would prefer that I pivot toward using the event system, I think that would work for me over making the hooks cancel-able. I suppose I would just need to add events for those cases and listen to GM:TTT2AddedEvent to record the events in real time, right?

Reply:
Nah, I don't want to pivot you in that direction, I only wanted to make sure that you are aware of it. If you say that it doesn't work for you, then it is fine